src/libraries/Microsoft.PowerFx.Core/Public/FileLocation.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using Microsoft.PowerFx.Core.Errors;
+using Microsoft.PowerFx.Core.Localization;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Syntax
+{
+    /// <summary>
+    /// File-aware, Multi-line source span with Line and Column. 
+    /// Wheras <see cref="Span"/> is a character span within a single expression.
+    /// </summary>
+    public class FileLocation
```

Comment:
Span tracks within a single expression, like `character offset 17`
But we need to report within an overall file.  like `File foo.yaml, line 5, col 3`

FileLocation is the base of where the expression starts, and then we can add the Span to get the (file/line/col) to report to user.